src/early_multicellular_stage/editor/EarlyMulticellularEditor.cs

```diff
@@ -155,6 +149,48 @@ public override bool CancelCurrentAction()
         return false;
     }
 
+    public override void Redo()
+    {
+        CellType? cellType = null;
+
+        // We don't allow combining actions from different contexts,
+        // so we only need to check the first data for the cell type
+        var redoData = history.ActionToRedo()?.Data.FirstOrDefault();
```

Comment:
This seems like duplicated code with Undo, and I think also with the late multicellular editor I saw similar changes. Could this maybe be made into a generic method in the action history to check if the context is changed (and return the new context value)?